### DISCUSSION TITLE
NextJS Migration or Update (12.2.3 to higher version)

### DISCUSSION THREAD

**Discussion by VarnikaManvik** [vercel/next.js#67906] [Help]
### Summary

Hello Team,

I need assistance in updating my Next.js application from version 12.2.3 to the latest or highest version. My project currently uses React and React-DOM version 17.0.1.

However, when I attempt the update, the application does not function correctly due to numerous breaking changes and code breakages. My project is structured using the traditional page-based approach, not the latest app routing structure.

My goal is to update Next.js without making significant changes to the codebase or file structure. Any guidance or suggestions on how to achieve this would be greatly appreciated.

Thank you!

**Comment by little077** [comment]
What does a large number of significant changes refer to? I understand that following the documentation will not result in widespread errors, and your project has a certain degree of encapsulation

**Reply by VarnikaManvik** [reply]
it is asking to update react to >=18 and updating 17 to 18 requires lot of project level code changes too,
